diff --git a/src/graphics/d3d12/command_processor.cpp b/src/graphics/d3d12/command_processor.cpp
index 19874a5..ac034cf 100644
--- a/src/graphics/d3d12/command_processor.cpp
+++ b/src/graphics/d3d12/command_processor.cpp
@@ -10,6 +10,7 @@
  */
 
 #include <algorithm>
+#include <bit>
 #include <cstdarg>
 #include <cstring>
 #include <sstream>
@@ -4115,9 +4116,9 @@ bool D3D12CommandProcessor::UpdateBindings(const D3D12Shader* vertex_shader,
     }
     for (uint32_t i = 0; i < 4; ++i) {
       uint64_t float_constant_map_entry = float_constant_map_vertex.float_bitmap[i];
-      uint32_t float_constant_index;
-      while (rex::bit_scan_forward(float_constant_map_entry, &float_constant_index)) {
-        float_constant_map_entry &= ~(1ull << float_constant_index);
+      while (float_constant_map_entry) {
+        uint32_t float_constant_index = std::countr_zero(float_constant_map_entry);
+        float_constant_map_entry &= float_constant_map_entry - 1;
         std::memcpy(
             float_constants,
             &regs[XE_GPU_REG_SHADER_CONSTANT_000_X + (i << 8) + (float_constant_index << 2)],
@@ -4141,9 +4142,9 @@ bool D3D12CommandProcessor::UpdateBindings(const D3D12Shader* vertex_shader,
           pixel_shader->constant_register_map();
       for (uint32_t i = 0; i < 4; ++i) {
         uint64_t float_constant_map_entry = float_constant_map_pixel.float_bitmap[i];
-        uint32_t float_constant_index;
-        while (rex::bit_scan_forward(float_constant_map_entry, &float_constant_index)) {
-          float_constant_map_entry &= ~(1ull << float_constant_index);
+        while (float_constant_map_entry) {
+          uint32_t float_constant_index = std::countr_zero(float_constant_map_entry);
+          float_constant_map_entry &= float_constant_map_entry - 1;
           std::memcpy(
               float_constants,
               &regs[XE_GPU_REG_SHADER_CONSTANT_256_X + (i << 8) + (float_constant_index << 2)],
